File: eclipse_old/monte_carlo.php
<?include 'header.php'?>


<div id="eclipse_main">

	<div id="eclipse_center">
		<form id="monte_carlo" method="GET" action="monte.php">
			<table align="center">
				<tr><td colspan=2 align="center"><b>Please enter the following values</b></td></tr>
				<tr>
					<td>Name</td><td><input type="text" name="s_name" value="RIMM" /></td>
				</tr>
				<tr>
					<td>S0</td><td><input type="text" name="s_zero" value=16 /></td>
				</tr>
				<tr>
					<td>R</td><td><input type="text" name="rounds" value=1000 /></td>
				</tr>
				<tr>
					<td>t</td><td><input type="text" name="t" value=1 /></td>
				</tr>
				<tr>
					<td>miu</td><td><input type="text" name="miu" value=0.1 /></td>
				</tr>
				<tr>
					<td>sigma</td><td><input type="text" name="sd" value=0.02 /></td>
				</tr>
				<tr>
					<td>alpha</td><td><input type="text" name="alpha" value=0.2 /></td>
				</tr>
				<tr>
					<td>theta</td><td><input type="text" name="theta" value=0.02 /></td>
				</tr>
				<tr>
					<td><input type="submit" value="Simulate!"></td>
					<td><input type="reset" value="Reset Fields"></td>
				</tr>
			</table>
		</form>
	</div>
	
	<div id="eclipse_right">
		<table align="center">
			<tr><td><i>Instruction</i></td></tr>
			<tr><td><b>Name:</b> Name or ticker of your stock</td></tr>
			<tr><td><b>S0:</b> Stock price at time 0</td></tr>
			<tr><td><b>R:</b> Number of simulations you want</td></tr>
			<tr><td><b>t:</b> time-units after current time(S1,S2..)</td></tr>
			<tr><td><b>miu:</b> Lognormal distribution mean</td></tr>
			<tr><td><b>sigma:</b> Lognormal distribution standard deviation</td></tr>
			<tr><td><b>alpha:</b> the stock's continuously annual compounded rate of return</td></tr>
			<tr><td><b>theta:</b> the stock's continuous annual divident rate</td></tr>
			<tr><td>All fields can be entered single value(eg. 40) or comma seperated values(eg. 40,50,40)</td></tr>
		</table>
	</div>
	
<!--	
NASDAQ:	JKHY
NASDAQ:	SLP
NASDAQ:	MLAB
FDS
FEIC
NASDAQ: ASMI
NASDAQ: MELI
NASDAQ: SYNT
NYSE:	MR

Final:
ASMI
EBIX
FEIC
MELI
MLAB
MR
SLP
SYNT
-->
</div>


<?include 'footer.php'?>

File: eclipse_old/models/monte_carlo/monte_carlo.py
import random as rand
import math as m

class Monte_carlo:
	'''
	Two usage of monte_carlo function:
	first	S0, round, t, miu, sigma				5 parameters
	second	S0, round, t, alpha, theta, sigma		6 parameters
	
	S0: starting price of stock
	round: number of runs for monte carlo
	t: depth of forecasting
	miu: mean of lognormal distribution
	sigma: standard deviation of lognormal distribution
	
	alpha: the stock's continuously annual compounded rate of return
	theta: the stock's continuous annual divident rate
	
	there are two methods to do calculation
	
	'''
	S0 = 40
	round = 1000
	t = 1
	miu = 3
	sigma = 4
	
	alpha = 0.2
	theta = 0.02
	def __init__(self, *args, **kwargs):
		if len(args) == 5:
			self.S0 = args[0]
			self.round = args[1]
			self.t = args[2]
			self.miu = args[3]
			self.sigma = args[4]
			
		if len(args) == 6:
			self.S0 = args[0]
			self.round = args[1]
			self.t = args[2]
			self.alpha = args[3]
			self.theta = args[4]
			self.sigma = args[5]
			self.miu = self.alpha - self.theta - self.sigma**2 * 0.5
			
	def __repr__(self):
		return 'S0: %s; R: %s; t: %s; miu: %s; sigma: %s; alpha: %s; theta: %s;' % (self.S0, self.round, self.t, self.miu, self.sigma, self.alpha, self.theta)
		
	def __str__(self):
		return 'S0: %s; R: %s; t: %s; miu: %s; sigma: %s; alpha: %s; theta: %s;' % (self.S0, self.round, self.t, self.miu, self.sigma, self.alpha, self.theta)
			
	def rand_lognormal_gen(self):
		z = 0
		for i in range(12):
			z = z + rand.uniform(0,1)
		z = z - 6
		n = self.miu + self.sigma*z
		x = m.exp(n)
		return x
		
	def simulate_once(self):
		
		rlt = []
		s = self.S0
		for i in range(self.t):
			s = s * self.rand_lognormal_gen()
			rlt.append(s)
		
		return rlt
	
	def simulate_complete(self):
		rlt = []
		s = []
		for i in range(self.round):
			s = self.simulate_once()
			rlt.append(s)
			
		return rlt